app/debug.conf

```diff
@@ -0,0 +1,23 @@
+# Copyright (c) 2021 Nordic Semiconductor ASA
+# SPDX-License-Identifier: Apache-2.0
+#
+# This file can be used as an overlay to enable debug-related Kconfig options
+# in the application. See the README for more details.
+
+# compiler
+CONFIG_DEBUG_OPTIMIZATIONS=y
+
+# console
+CONFIG_CONSOLE=y
+
+# UART console
+CONFIG_SERIAL=y
+CONFIG_UART_CONSOLE=y
+
+# segger RTT console (uncomment if using SEGGER)
+# CONFIG_USE_SEGGER_RTT=y
+# CONFIG_RTT_CONSOLE=y
```

Comment:
Instead of uncommenting, wouldn't it be better to place this into its own fragment ?
Such as a `rtt.conf` file:
```
# segger RTT console
CONFIG_USE_SEGGER_RTT=y
CONFIG_RTT_CONSOLE=y
```

Users can then do `-DOVERLAY_CONFIG=debug.conf` for just debug or `-DOVERLAY_CONFIG="debug.conf;rtt.conf"`  for debug with rtt.

And no need for modifying code under version control when building.

Reply:
I like the fragment approach.  In a prior job we found a similar need for composability and ended up having several layers of composition.  Too bad we have to do so much on the command line.  It would be more convenient for me to have the base `OVERLAY_CONFIG` value be set in the project's `CMakeLists.txt` and use `-DOVERLAY_CONFIG` for alternate test configurations.

Reply:
done

Reply:
> I like the fragment approach. In a prior job we found a similar need for composability and ended up having several layers of composition. Too bad we have to do so much on the command line. It would be more convenient for me to have the base OVERLAY_CONFIG value be set in the project's CMakeLists.txt and use -DOVERLAY_CONFIG for alternate test configurations.

If it should always be set, then why not just place those settings in the `prj.conf` ?
Or if you want multiple build types, a `prj.conf` and `prj_debug.conf`.

Bonus info, the CMake presets introduced with CMake 3.19 looks very promising in order to make it easier to provide various build combinations that can also be integrated with IDEs.